client/dashboard/app/dev-tools/preferences.tsx

```diff
@@ -6,39 +6,174 @@ import {
+	return value;
+};
+
+const renderPreference = ( name: string, value: any ) => {
+	if ( typeof value === 'string' ) {
+		return <EditablePreference inputType="string" name={ name } value={ value } />;
+	}
+
+	if ( typeof value === 'boolean' ) {
+		return <EditablePreference inputType="checkbox" name={ name } value={ value } />;
+	}
+
+	if ( typeof value === 'number' ) {
+		return <EditablePreference inputType="number" name={ name } value={ value } />;
+	}
+
+	if ( Array.isArray( value ) ) {
+		return <ArrayPreference value={ value } />;
+	}
+
+	return null;
+};
+
+function ArrayPreference( { value }: { value: InputValue[] } ) {
+	return (
+		<ul>
+			{ value.map( ( preference, index ) => (
+				<li key={ index }>{ JSON.stringify( preference ) }</li>
+			) ) }
+		</ul>
+	);
+}
+
+function EditablePreference( {
+	inputType,
+	name,
+	value,
+}: {
+	inputType: InputType;
+	name: string;
+	value: InputValue;
+} ) {
+	const { mutate: savePreference } = useMutation(
 		userPreferenceMutation( name as keyof UserPreferences )
 	);
+	const [ formData, setFormData ] = useState( { [ name ]: value } );
+
+	const handleSave = () => {
+		savePreference( formData[ name ] );
+	};
 
-	const handleClick = () => {
-		unsetPreference( null as unknown as UserPreferences[ keyof UserPreferences ] );
+	const handleReset = () => {
+		setFormData( { [ name ]: value } );
 	};
 
+	const fields: Field< { [ name ]: InputValue } >[] = useMemo(
+		() => [
+			{
+				id: name,
+				Edit: ( { field, onChange, data } ) => {
+					const { id, getValue } = field;
+					if ( inputType === 'checkbox' ) {
+						return (
+							<CheckboxControl
+								__nextHasNoMarginBottom
+								checked={ getValue( { item: data } ) }
+								onChange={ ( newValue ) =>
+									onChange( { [ id ]: formattedValue( inputType, newValue ) } )
+								}
+							/>
+						);
+					}
+
+					return (
+						<InputControl
+							type={ inputType }
+							value={ getValue( { item: data } ) }
+							size="small"
+							onChange={ ( newValue ) =>
+								onChange( { [ id ]: formattedValue( inputType, newValue ) } )
+							}
+							hideLabelFromVision
+						/>
+					);
+				},
+			},
+		],
+		[ name, inputType ]
+	);
+
+	const isDirty = formData[ name ] !== value;
+
 	return (
-		<div>
+		<VStack>
+			<DataForm< { [ name ]: InputValue } >
+				data={ formData }
+				fields={ fields }
+				form={ {
+					layout: { type: 'regular' as const },
+					fields: [ name ],
+				} }
+				onChange={ ( edits ) => {
+					setFormData( ( current ) => ( { ...current, ...edits } ) );
+				} }
+			/>
+			<HStack justify="flex-start" spacing={ 1 }>
+				<Button variant="primary" size="small" disabled={ ! isDirty } onClick={ handleSave }>
+					{ __( 'Save' ) }
+				</Button>
+				<Button variant="secondary" size="small" disabled={ ! isDirty } onClick={ handleReset }>
+					{ __( 'Reset' ) }
+				</Button>
+			</HStack>
+		</VStack>
+	);
+}
+
+function Preference( { name, value }: { name: string; value: any } ) {
```

Comment:
These two `any`s can be `unknown`s and it fixes the lint warning.

```suggestion
function Preference( { name, value }: { name: string; value: unknown } ) {
```

Reply:
Sounds good 👍  Updated.